### ISSUE TITLE
"Train Until" functionality

### ISSUE THREAD

**Issue by shashi-netra** [ultralytics/yolov5#440]
## 🚀 Feature

This is common now in many ML training tasks, to train until `val loss` stops falling over say n (=10) epochs.
I suspect this should be an easy feature to add. Wish I had the Torch coding chops, otherwise would have sent a PR.

## Motivation
Figuring out the right num of epochs is a confusing and painful experience.

**Comment by glenn-jocher** [comment]
@shashi-netra train using default settings, what's so confusing about it?

You can modify the scheduler for whatever you want. See 
https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.ReduceLROnPlateau
